File: FizzBuzz.scala
object FizzBuzz {
	def fizzBuzz(fizz : Int, buzz : Int, num : Int) : String = 
		if (num % fizz == 0 && num % buzz == 0) "FizzBuzz"
		else if (num % fizz == 0) "Fizz"
		else if (num % buzz == 0) "Buzz"
		else num + ""

	def main(args: Array[String]) {
		for ( item <- 1 to 100) {
			println(fizzBuzz(3, 5, item));
		}
	}
}

File: fizzbuzz.erl
-module(fizzbuzz).
-export([fizzbuzz/3]).

fizzbuzz(fizz,buzz,num) ->
	res = if (num rem fizz) == 0, (num rem buzz) == 0 -> "FizzBuzz~n";
		(num rem fizz) == 0 -> "Fizz~n";
		(num rem buzz) == 0 -> "Buzz~n";
		true -> num
	end,
	{res}.
